src/Paket.Core/RemoteDownload.fs

```diff
@@ -189,7 +239,35 @@ let downloadRemoteFiles(remoteFile:ResolvedSourceFile,destination) = async {
 }
 
 let DownloadSourceFiles(rootPath, groupName, force, sourceFiles:ModuleResolver.ResolvedSourceFile list) =
-    sourceFiles
+    let remoteFiles,gitRepos = 
+        sourceFiles
+        |> List.partition (fun x -> match x.Origin with | GitLink _ -> false | _ -> true)
+
+    let gitDownloads =
+        gitRepos
+        |> List.map (fun gitRepo ->
+            async {
+                let destination = gitRepo.FilePath(rootPath,groupName)
+                let repoFolder = Path.Combine(destination,gitRepo.Project)
+                let isInCorrectVersion =
+                    if Directory.Exists repoFolder then
+                        try
+                            let hash = Git.CommandHelper.runSimpleGitCommand repoFolder ("rev-parse head")
+                            hash = gitRepo.Commit
+                        with
+                        | _ ->
+                            // something is wrong with the repo - recreate it
+                            Directory.Delete(repoFolder,true)
+                            false
+                    else false
+
+                if isInCorrectVersion then
```

Comment:
Shouldn't it be `if not force && isInCorrectVersion then` ? Just like for the other origins.